apps/desktop/desktop_native/core/src/secure_memory/secure_key/mod.rs

```diff
@@ -50,60 +88,93 @@ impl SecureKeyContainer for CrossPlatformSecureKeyContainer {
             if keyctl::KeyctlSecureKeyContainer::is_supported() {
+                info!("Using keyctl for secure key storage");
                 return CrossPlatformSecureKeyContainer::Keyctl(
                     keyctl::KeyctlSecureKeyContainer::from_key(key),
                 );
             }
         }
+
+        // Falling back to mlock means that the key is accessible via memory dumping.
+        info!("Falling back to mlock for secure key storage");
         CrossPlatformSecureKeyContainer::Mlock(mlock::MlockSecureKeyContainer::from_key(key))
     }
 
     fn is_supported() -> bool {
+        // Mlock is always supported as a fallback.
         true
     }
 }
 
-/// An ephemeral key that is protected using a platform mechanism. It is generated on construction freshly, and can be used
-/// to encrypt and decrypt segments of memory. Since the key is ephemeral, persistent data cannot be encrypted with this key.
-/// On Linux and Windows, in most cases the protection mechanisms prevent memory dumps/debuggers from reading the key.
-#[allow(unused)]
-pub(crate) struct SecureMemoryEncryptionKey(CrossPlatformSecureKeyContainer);
-
-impl SecureMemoryEncryptionKey {
-    pub fn new() -> Self {
-        let key = crypto::MemoryEncryptionKey::new();
-        let container = CrossPlatformSecureKeyContainer::from_key(key);
-        SecureMemoryEncryptionKey(container)
-    }
-
-    #[allow(unused)]
-    pub fn encrypt(&self, plaintext: &[u8]) -> crypto::EncryptedMemory {
-        let key = self.0.as_key();
-        key.encrypt(plaintext)
-    }
-
-    #[allow(unused)]
-    pub fn decrypt(&self, encrypted: &crypto::EncryptedMemory) -> Vec<u8> {
-        let key = self.0.as_key();
-        key.decrypt(encrypted)
+fn get_env_forced_container() -> Option<CrossPlatformSecureKeyContainer> {
+    let env_var = std::env::var("SECURE_KEY_CONTAINER_BACKEND");
+    match env_var.as_deref() {
+        #[cfg(target_os = "windows")]
+        Ok("dpapi") => {
+            info!("Forcing DPAPI secure key container via environment variable");
+            Some(CrossPlatformSecureKeyContainer::Dpapi(
+                dpapi::DpapiSecureKeyContainer::from_key(crypto::MemoryEncryptionKey::new()),
+            ))
+        }
+        #[cfg(target_os = "linux")]
+        Ok("memfd_secret") => {
+            info!("Forcing memfd_secret secure key container via environment variable");
+            Some(CrossPlatformSecureKeyContainer::MemfdSecret(
+                memfd_secret::MemfdSecretSecureKeyContainer::from_key(
+                    crypto::MemoryEncryptionKey::new(),
+                ),
+            ))
+        }
+        #[cfg(target_os = "linux")]
+        Ok("keyctl") => {
+            info!("Forcing keyctl secure key container via environment variable");
+            Some(CrossPlatformSecureKeyContainer::Keyctl(
+                keyctl::KeyctlSecureKeyContainer::from_key(crypto::MemoryEncryptionKey::new()),
+            ))
+        }
+        Ok("mlock") => {
```

Comment:
Shoudln't mlock be linux specific ?
```suggestion
        #[cfg(target_os = "linux")]
        Ok("mlock") => {
```